src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -270,6 +274,12 @@ ResetShardPlacementTransactionState(void)
 	 * round.
 	 */
 	shardConnectionHash = NULL;
+
+	if (MultiShardCommitProtocol == COMMIT_PROTOCOL_BARE)
+	{
+		MultiShardCommitProtocol = SavedMultiShardCommitProtocol;
+		SavedMultiShardCommitProtocol = COMMIT_PROTOCOL_BARE;
```

Comment:
If we had some `INVALID` value in the enum I was going to use it, but didn't feel like adding it myself. Leaving as-is.